Update selection-statements.md - fixed example meaning, highlighted difference between switch statement and switch expression

theluckyprogrammer opened the pull request:
## Summary

Update selection-statements.md - fixed example meaning, highlighted the difference between a switch statement and switch expression

Fixes #Issue_Number (if available)


<!-- PREVIEW-TABLE-START -->

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/language-reference/statements/selection-statements.md](https://github.com/dotnet/docs/blob/26598dba85d060c6c251ca2cda400d4923506cc4/docs/csharp/language-reference/statements/selection-statements.md) | [Selection statements - `if`, `if-else`, and `switch`](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/statements/selection-statements?branch=pr-en-us-44945) |


<!-- PREVIEW-TABLE-END -->

theluckyprogrammer committed:
Update selection-statements.md - fixed example meaning, highlighted difference between switch statement and switch expression

Update selection-statements.md - fixed example meaning, highlighted difference between switch statement and switch expression

BillWagner commented on docs/csharp/language-reference/statements/selection-statements.md:
```diff
@@ -64,9 +64,16 @@ You can specify multiple case patterns for one section of a `switch` statement,
 :::code language="csharp" source="snippets/selection-statements/SwitchStatement.cs" id="MultipleCases":::
 
-Within a `switch` statement, control can't fall through from one switch section to the next. As the examples in this section show, typically you use the `break` statement at the end of each switch section to pass control out of a `switch` statement. You can also use the [return](jump-statements.md#the-return-statement) and [throw](exception-handling-statements.md#the-throw-statement) statements t ...
+Within a `switch` statement, control can fall from one switch section to the next. As the examples in this section show, typically you use the `break` statement at the end of each switch section to pass control out of a `switch` statement. You can also use the [return](jump-statements.md#the-return-statement) and [throw](exception-handling-statements.md#the-throw-statement) statements to pass con ...
```
This is incorrect.  See [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ+WRiqQLIAUAlgHYAupHgFcAtgEpa9RtIDOAdy58wAC1Idh4qdIZ0C2/aTABDGTFI4QWg/ow4AnBwBEEHjEdiA3FevHTpLJZ61ga2Do58chDuXkH6AL5WCQRxQA=). In C#, without a `break` statement, the compiler generates an error. If you want to fall through to the following case, you need an explicit `goto`:  [Sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ+WRiqQLIAUAlgHYAupHgFcAtgEpa9RtIDOAdy58wAC1Idh4qdIZ0C2/aTABDGTFI4QWg/ow4AnBwBEEHjEdiA3FevSA5hD4IQxMzLC89HwZjU1IsSwjI6VsHRz45CHdwxO1gWCMAayyDAF8rUoJioA=)

theluckyprogrammer replied:
I had in mind such case
```
public static void M(int num) {
        switch (num)
        {
            case 1:
            case 2:
                Console.WriteLine("two");
                break;
        }
```

In above case, it can fall without break; and will not throw exception. 

In switch expression, it will look like:
```
public static void M(int num)
{
    string result = num switch
    {
        1 or 2 => "two",
        _ => "other"
    };

    Console.WriteLine(result);
}
```

BillWagner commented on docs/csharp/language-reference/statements/selection-statements.md:
```diff
@@ -64,9 +64,16 @@ You can specify multiple case patterns for one section of a `switch` statement,
+> [!IMPORTANT]
+> Differences between **switch expression** and **switch statement**:
+> - **switch expression** must return a value, a **switch statement** does not. <br>
+> - **switch expressions** don't require break statements - fallthrough is not allowed.
```
This is incorrect (see above)

theluckyprogrammer replied:
now I see that the statement that switch expression must return a value is really unfortunate; it is rather as follows: "case must have an explicitly set instruction."

BillWagner commented on docs/csharp/language-reference/statements/selection-statements.md:
```diff
@@ -64,9 +64,16 @@ You can specify multiple case patterns for one section of a `switch` statement,
+> Differences between **switch expression** and **switch statement**:
+> - **switch expression** must return a value, a **switch statement** does not. <br>
+> - **switch expressions** don't require break statements - fallthrough is not allowed.
+> - **switch expression** is typically used in contexts of value return and value assignment, often as [expression-bodied members](../../programming-guide/statements-expressions-operators/expression-bodied-members.md). <br> 
```
This is true, but I'm not sure how useful it is. Why did you want to add this?

theluckyprogrammer replied:
@BillWagner what do you think?

theluckyprogrammer replied:
I've updated this file with a more precise description, based on the existing switch statement example. There was the case that I wanted to include ( empty case section )

BillWagner reviewed (changes requested):
Hi @theluckyprogrammer 

There are a few technical errors here that need to be corrected before we can merge this. I like the direction you're taking with the bullet list to compare the `switch` statement and the `switch` expression. I've got some questions there, but that could be updated and merged.

theluckyprogrammer committed:
Update selection-statements.md

Included is a description of intentional case sections that fall back if they share the same or interoperable logic.

BillWagner commented on docs/csharp/language-reference/statements/selection-statements.md:
```diff
@@ -66,11 +66,21 @@ You can specify multiple case patterns for one section of a `switch` statement,
 
-In an expression context, you can use the [`switch` expression](../operators/switch-expression.md) to evaluate a single expression from a list of candidate expressions based on a pattern match with an expression.
+> [!IMPORTANT]
+> Missing `break` statement will throw exception, however **empty** case sections, like `case < 0:` in example above, **are** permitted to fall through. This deliberate design choice allows for concisely handling multiple cases that share the same or interdependent logic.
```
One small nit: It's a compiler error, not an exception:

```suggestion
> Every *switch section* must end with a `break`, `goto` or `return`. Falling through from one switch section to the next generates a compiler error. However, multiple switch labels can be applied to the same switch section, like `case < 0:` in example above. This deliberate design choice allows for concisely handling multiple cases that share the same or interdependent logic.
```

BillWagner reviewed (approved):
This looks good now @theluckyprogrammer 

I had some verbiage suggestions to make, and then I'll :shipit:

BillWagner committed:
Update docs/csharp/language-reference/statements/selection-statements.md

BillWagner commented on docs/csharp/language-reference/statements/selection-statements.md:
```diff
@@ -66,11 +66,21 @@ You can specify multiple case patterns for one section of a `switch` statement,
+In an expression context, you can use the [`switch` expression](../operators/switch-expression.md) to evaluate a single expression from a list of candidate expressions based on a pattern match with an expression.<br>
+
+> [!IMPORTANT]
+> Differences between **switch expression** and **switch statement**:
+> - **switch statement** is used to control the execution flow within a block of code.<br>
+> - **switch expression** is typically used in contexts of value return and value assignment, often as a [expression-bodied members](../../programming-guide/statements-expressions-operators/expression-bodied-members.md). <br> 
+> - a **switch expression** case section cannot be empty, a **switch statement** can.<br>
+
```
Fix multiple lint issues:

```suggestion
> Differences between **switch expression** and **switch statement**:
>
> - **switch statement** is used to control the execution flow within a block of code.
> - **switch expression** is typically used in contexts of value return and value assignment, often as a [expression-bodied members](../../programming-guide/statements-expressions-operators/expression-bodied-members.md).
> - a **switch expression** case section cannot be empty, a **switch statement** can.
```

BillWagner committed:
Update docs/csharp/language-reference/statements/selection-statements.md
